lib/bolt/transport/ssh/exec_connection.rb

```diff
@@ -12,8 +12,12 @@ def initialize(target)
           raise Bolt::ValidationError, "Target #{target.safe_name} does not have a host" unless target.host
 
           @target = target
-          ssh_config = Net::SSH::Config.for(target.host)
-          @user = @target.user || ssh_config[:user] || Etc.getlogin
+          unless target.transport_config['native-ssh']
```

Comment:
We only ever use this transport if `native-ssh` is true for a target, so we should be able to skip the `if` and just use `@user = @target.user || Etc.getlogin` directly.